src/v/iceberg/manifest_entry_values.cc

```diff
@@ -204,7 +206,8 @@ data_file data_file_from_value(struct_value v) {
     file.column_sizes = get_counts_map(std::move(fs[6]));
     file.value_counts = get_counts_map(std::move(fs[7]));
     file.null_value_counts = get_counts_map(std::move(fs[8]));
-    file.nan_value_counts = get_counts_map(std::move(fs[9]));
+    file.distinct_counts = get_counts_map(std::move(fs[9]));
+    file.nan_value_counts = get_counts_map(std::move(fs[10]));
```

Comment:
where is the code that determines what index something in v.fields is at (ie fs[i])?

Reply:
The ordering here must match the schema created by the code in manifest_entry_type.cc